src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1146,6 +1157,145 @@ CATCH_RETURN();
+        UINT32 length;
+        RETURN_IF_FAILED(source->GetTextAtPosition(textPosition, &text, &length));
+
+        // Put it into a view for iterator convenience.
+        const std::wstring_view str(text, length);
+
+        // Find the first box drawing character in the string from the front.
+        const auto firstBox = std::find_if(str.cbegin(), str.cend(), _IsBoxDrawingCharacter);
+
+        // If we found no box drawing characters, move on with life.
+        if (firstBox == str.cend())
+        {
+            return S_OK;
+        }
+        // If we found one, keep looking forward until we find NOT a box drawing character.
+        else
+        {
+            // Find the last box drawing character.
+            const auto lastBox = std::find_if(firstBox, str.cend(), [](wchar_t wch) { return !_IsBoxDrawingCharacter(wch); });
+
+            // Skip distance is how far we had to move forward to find a box.
+            const auto firstBoxDistance = std::distance(str.cbegin(), firstBox);
+            UINT32 skipDistance;
+            RETURN_HR_IF(E_ABORT, !base::MakeCheckedNum(firstBoxDistance).AssignIfValid(&skipDistance));
+
+            // Move the position/length of the outside counters up to the part where boxes start.
+            textPosition += skipDistance;
+            textLength -= skipDistance;
+
+            // Run distance is how many box characters in a row there are.
+            const auto runDistance = std::distance(firstBox, lastBox);
+            UINT32 mappedLength;
+            RETURN_HR_IF(E_ABORT, !base::MakeCheckedNum(runDistance).AssignIfValid(&mappedLength));
+
+            // Split the run and set the box effect on this segment of the run
+            RETURN_IF_FAILED(_SetBoxEffect(textPosition, mappedLength));
+
+            // Move us forward for the outer loop to continue scanning after this point.
+            textPosition += mappedLength;
+            textLength -= mappedLength;
+        }
+    }
+
+    return S_OK;
+}
+CATCH_RETURN();
+
+// Routine Description:
+// - A callback to split a run and apply box drawing characteristics to just that sub-run.
+// Arguments:
+// - textPosition - the index to start the substring operation
+// - textLength - the length of the substring operation
+// Return Value:
+// - S_OK or appropriate STL/GSL failure code.
+[[nodiscard]] HRESULT STDMETHODCALLTYPE CustomTextLayout::_SetBoxEffect(UINT32 textPosition,
+                                                                        UINT32 textLength)
+try
+{
+    _SetCurrentRun(textPosition);
+    _SplitCurrentRun(textPosition);
+
+    while (textLength > 0)
+    {
+        auto& run = _FetchNextRun(textLength);
+
+        if (run.fontFace == _font)
+        {
+            run.drawingEffect = _boxDrawingEffect;
+        }
+        else
+        {
+            ::Microsoft::WRL::ComPtr<IBoxDrawingEffect> eff;
+            RETURN_IF_FAILED(s_CalculateBoxEffect(_format.Get(), run.fontFace.Get(), &eff, run.fontScale));

```

Comment:
clever. recalculating it if the font isn't what we expect.

This will come at a perf hit _per run_, right?

Reply:
Yes, it is a perf hit per run. I want to make a cache of them and also to cache the base font's metrics at the time the font is selected. I'll get that in an upcoming rev to this PR.

Reply:
Ehhhh we verbally talked about this and aren't going to do anything about it for now.